eosdis-nasa/earthdata-pub-api
# Edpub 1297 split uploads

# Description

Back end changes for splitting uploads into sample/documentation buckets

## Spec

See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1297

Related PR: https://github.com/eosdis-nasa/earthdata-pub-dashboard/pull/86

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches to SIT
3. Log into EDPUB and get authentication information
4. Authenticate in swagger
5. Scroll down to `api/data/upload/getPostUrl`
6. Use the Try it Out feature to verify endpoint returns expected result
7. Navigate to the request overview page for a new or existing request (*from this step on is a duplicate of the validation for the related PR).
8. Select documentation and click upload to select a file
9. Verify that the file shows the correct category in the table after upload
10. Navigate to the DAR form
11. Verify previously uploaded file shows up in the documentation section
12. Attempt to save the form and verify that a red box appears around the sample file question
13. Upload a file to the sample files
14. Save the form and verify that the red box has disappeared

---

## Change Log

* Added upload split by category
